app/controllers/one_login_controller.rb

```
@@ -90,20 +90,8 @@ def sign_out_complete
   end
 
   def failure
-    # This action will catch all OAuth failures
-    # from all strategies i.e. OneLogin and DfE Sign-in
-    return unless params[:strategy] == 'one_login'
-
-    session_error = SessionError.create!(
-      body: "One login failure with #{params[:message]}",
-    )
-    Sentry.capture_message(
-      "#{session_error.body}, check session_error record #{session_error.id}",
-      level: :error,
-    )
-
-    session[:session_error_id] = session_error.id if session_error.present?
-    redirect_to auth_one_login_sign_out_path
+    terminate_session
+    redirect_to internal_server_error_path
```

Are we sure we don't want to redirect to `auth_one_login_sign_out_path`? or maybe (/)? So far, the experience has been really seamless for the user because we have just given them an opportunity to login again. This would mean they land on a 500 page -- we get to ignore the error, but now they don't, right?

---

Good point - the problem just now is that this endpoint is capturing Users from OneLogin and DfE Sign in. Redirecting to `auth_one_login_sign_out_path` will ultimately redirect to `candidate_interface_create_account_or_sign_in_path` that doesn't feel right for Providers 🤔
Not sure how best to handle this. Ideally we'd have separate endpoints for each strategy ie. `auth/dfe/failure` and `auth/one-login/failure` rather than the generic `auth/failure` that we currently have 🤔
Looking at the OmniAuth config I assume the `path_prefix` should make this possible but I've had a hard time finding documentation - it looks like it uses the `path_prefix` [here](https://github.com/omniauth/omniauth/blob/e23567ac860f4adfb425db226d68a03235078941/lib/omniauth/failure_endpoint.rb#L31)

I think for now we can redirect to `root_path` as you suggested 👌